common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -1864,35 +1871,27 @@ private void finalizePartition() throws IOException {
       metaFile.getChannel().truncate(metaFile.getPos());
     }
 
-    void closeAllFilesAndDeleteIfNeeded(boolean delete) {
+    private void deleteAllFiles() {
       try {
-        if (dataChannel.isOpen()) {
-          dataChannel.close();
-        }
-        if (delete) {
-          dataFile.delete();
-        }
-      } catch (IOException ioe) {
-        logger.warn("Error closing data channel for {} reduceId {}",
+        if (!dataFile.delete()) {
+          logger.warn("Error deleting data file for {} reduceId {}",
             appAttemptShuffleMergeId, reduceId);
+        }
+      } catch (Exception exception) {
+        logger.warn("Error deleting data file for {} reduceId {}",
+          appAttemptShuffleMergeId, reduceId);
       }
       try {
-        metaFile.close();
-        if (delete) {
-          metaFile.delete();
-        }
+        metaFile.delete();
       } catch (IOException ioe) {
-        logger.warn("Error closing meta file for {} reduceId {}",
-            appAttemptShuffleMergeId, reduceId);
-        }
+        logger.warn("Error deleting meta file for {} reduceId {}",
+          appAttemptShuffleMergeId, reduceId);
+      }
```

Comment:
Drop all `IOException` catch blocks in this method - they are not thrown

Reply:
org.apache.spark.network.shuffle.RemoteBlockPushResolver.MergeShuffleFile#delete throw IOException, if drop it in this method, the delete method need modify.

Reply:
Thanks, drop them